pytest-saas/test/integration/pytest_saas_test.py

```diff
@@ -0,0 +1,116 @@
+import os
+import re
+from inspect import cleandoc
+from unittest import mock
+
+import pytest
+
+pytest_plugins = "pytester"
+
+
+@pytest.fixture
+def make_test_files():
+    def make(pytester, files):
+        pytester.makepyfile(**files)
+
+    return make
+
+
+def _testfile(body):
+    testname = re.sub(r"^.*def ([^(]+).*", "\\1", body, flags=re.S)
+    return { testname: cleandoc(body) }
+
+
+def _cli_args(*args):
+    return args
+
+
+def _env(**kwargs):
+    return kwargs
+
+
+@pytest.mark.parametrize(
+    "files,cli_args",
+    [
+        ( _testfile("""
+          def test_no_cli_args(request):
+              assert not request.config.getoption("--keep-saas-database")
+              assert request.config.getoption("--saas-database-id") is None
+          """),
+          _cli_args(),
+         ),
+        ( _testfile("""
+          import os
+          def test_cli_args(request):
+              assert request.config.getoption("--keep-saas-database")
+              assert "123" == request.config.getoption("--saas-database-id")
+              assert "PST" == request.config.getoption("--project-short-tag")
+          """),
+          _cli_args(
+              "--keep-saas-database",
+              "--project-short-tag", "PST",
+              "--saas-database-id", "123",
+          ),
+         ),
```

Comment:
What is the intention of these tests. Do want to test, if the CLI options are defined?

Reply:
Yes. I took a look at the pytest-exasol-itde plugin and created similar test cases.
This is also based on a discussion with @Nicoretti.

Reply:
see next push